src/test/java/org/opensearch/neuralsearch/query/NeuralQueryIT.java

```diff
@@ -48,6 +49,13 @@ public void setUp() throws Exception {
         modelId.compareAndSet(modelId.get(), prepareModel());
     }
 
+    @After
+    @SneakyThrows
+    public void tearDown() {
+        super.tearDown();
```

Comment:
this is required to minimize chance of model not being deployed due to open memory CB, this happens in case we leave model from previous test case. We use new model for every test, and old model can be undeployed and deleted to free resources after each test case execution.

Reply:
Shall we add this explanation as comments for future reference?

Reply:
ack